lib/BitcastUtils.cpp

```diff
@@ -831,25 +834,53 @@ bool RemoveCstExprFromFunction(Function *F) {
   }
 
   bool Changed = !WorkList.empty();
-
   while (!WorkList.empty()) {
     auto *I = WorkList.back().first;
     auto OperandId = WorkList.back().second;
     WorkList.pop_back();
-
+    IRBuilder<> B(I);
     auto InsertBefore = I;
     if (auto phi = dyn_cast<PHINode>(I)) {
       InsertBefore =
           phi->getIncomingBlock(phi->getIncomingValueNumForOperand(OperandId))
               ->getFirstNonPHI();
     }
 
-    auto Operand = dyn_cast<ConstantExpr>(I->getOperand(OperandId))
-                       ->getAsInstruction(InsertBefore);
-    CheckInstruction(Operand);
-    I->setOperand(OperandId, Operand);
-  }
+    if (auto CstStruct = dyn_cast<ConstantArray>(I->getOperand(OperandId))) {
+      unsigned numEle = CstStruct->getType()->getArrayNumElements();
+      Value *ArrayNew = UndefValue::get(CstStruct->getType());
+      for (unsigned i = 0; i < numEle; ++i) {
+        Value *Scalar = B.CreateExtractValue(CstStruct, i);
+        auto *ScalarInst =
```

Comment:
You need to test the result of `dyn_cast<ConstantExpr>` before trying to call `getAsInstruction` as the result of `dyn_cast` can be `nullptr`.
Also even if the cast does not work, something needs to be inserted in the array.